save-demo/src/main/kotlin/com/saveourtool/save/demo/service/DemoService.kt

```diff
@@ -88,9 +91,29 @@ class DemoService(
     )
 
     /**
-     * @return list of [Demo]s that are stored in database
+     * @param demoFilter [DemoFilter]
+     * @param pageSize amount of [Demo]s that should be fetched
+     * @return list of [Demo]s that match [DemoFilter]
      */
-    fun getAllDemos(): List<Demo> = demoRepository.findAll()
+    fun getFiltered(demoFilter: DemoFilter, pageSize: Int): Page<Demo> = demoRepository.findAll({ root, _, cb ->
+        with(demoFilter) {
+            val organizationNamePredicate = if (organizationName.isBlank()) {
+                cb.and()
```

Comment:
what's about status?

Reply:
Right now I check statuses after the filtered demo is fetched as I do not store the status in database due to no sync solution (right now there is no way to keep data up to date, need to implement some kind of service that does this task repeatedly over time). If you find it better, I could move status filtering to this method, but the point is that status check right now is an expensive operation

Reply:
Will fix that in the next PR.